File: 1/Sim.py
numTests = int(input())

for _ in range(numTests):
    output = []
    pos = 0
    line = input()

    for caracter in line:
        if caracter == '<':
            if pos-1 >= 0 and output:  # Comprueba que output no esté vacío
                output.pop(pos-1)
                pos -= 1
            continue

        elif caracter == '[':
            pos = 0
            continue

        elif caracter == ']':
            pos = len(output)

        else:
            output.insert(pos, caracter)
            pos += 1

    output = ''.join(output)
    print(output)


""" from collections import deque

numTests = int(input())

for _ in range(numTests):
    left = deque()
    right = deque()
    
    line = input()

    for caracter in line:
        if caracter == '<':
            if left:
                left.pop()
        elif caracter == '[':
            if left:
                right.extendleft(reversed(left))
                left.clear()
        elif caracter == ']':
            if right:
                left.extend(right)
                right.clear()
        else:
            left.append(caracter)

    left.extend(right)
    print(''.join(left)) """

File: 1/Sim.cpp
#include <iostream>
#include <list>
using namespace std;

int main() {
    int numTests;
    cin >> numTests;
    cin.ignore();  // Ignora el salto de línea

    for(int tests = 0; tests < numTests; tests++) {
        string line;
        getline(cin, line);

        list<char> output;
        auto pos = output.end();

        for(char c : line) {
            if(c == '<') {
                if(pos != output.begin()) {
                    pos--;
                    pos = output.erase(pos);
                }

            } else if(c == '[') {
                pos = output.begin();

            } else if(c == ']') {
                pos = output.end();

            } else {
                output.insert(pos, c);
            }
        }

        for(char c : output) {
            cout << c;
        }
        cout << endl;
    }

    return 0;
}